### PR TITLE
Add saving screenshots in subdirectories

### PR DESCRIPTION
**Pull request by SjorsO**
This PR adds the ability to save screenshots in subdirectories. Currently, the directory you give to the `takeScreenshot()` method has to exist, else it will throw an exception. With this PR you can save the screenshot to a directory that does not exist, the directory will be created automatically.

My use-case is that i'm working with Laravel Dusk, and i want to save screenshots for each form that the tests fill in. Currently, the screenshots all end up in the same directory, so i have to name them something like:

- `user-activation-single-company-step-1.png`
- `user-activation-single-company-step-2.png`
- `user-activation-multiple-companies-step-1.png`
- `user-activation-multiple-companies-step-2.png`
- (and many more)

With this PR i can save them in directories instead, like this:

- `user-activation/single-company/step-1.png`
- `user-activation/single-company/step-2.png`
- `user-activation/multiple-companies/step-1.png`
- `user-activation/multiple-companies/step-2.png`
- (and many more)

### CODE REVIEW

**Comment by SjorsO** [comment]
The CI fails, but the errors seem to be unrelated to my PR

File: tests/functional/RemoteWebDriverTest.php [hunk 1]
```diff
@@ -245,4 +245,27 @@ public function testShouldSaveScreenshotToFile()
...
+    public function testShouldSaveScreenshotToFileInSubDirectory()
+    {
+        if (!extension_loaded('gd')) {
+            $this->markTestSkipped('GD extension must be enabled');
+        }
+        if ($this->desiredCapabilities->getBrowserName() == WebDriverBrowserType::HTMLUNIT) {
+            $this->markTestSkipped('Screenshots are not supported by HtmlUnit browser');
+        }
+
+        $screenshotPath = sys_get_temp_dir() . '/fb-webdriver/dir/selenium-screenshot.png';
```
Comment (OndraM) [inline tests/functional/RemoteWebDriverTest.php]: To make sure the screenshot is actually saved into newly created directory, random directory and random file name could be used:

```suggestion
        $screenshotPath = sys_get_temp_dir() . '/' . uniqid('php-webdriver') . '/' . uniq('screenshot') '.png';
```

File: tests/functional/RemoteWebDriverTest.php [hunk 2]
```diff
@@ -245,4 +245,27 @@ public function testShouldSaveScreenshotToFile()
...
+
+        $this->driver->get($this->getTestPageUrl('index.html'));
+
+        $this->driver->takeScreenshot($screenshotPath);
+
+        $this->assertFileExists($screenshotPath);
+
+        unlink($screenshotPath);
```
Comment (OndraM) [inline tests/functional/RemoteWebDriverTest.php]: This does not remove the directory after the test, only the file. This should help:

```php
rmdir(dirname($screenshotPath));
```

File: tests/functional/RemoteWebDriverTest.php [hunk 3]
```diff
@@ -245,4 +245,27 @@ public function testShouldSaveScreenshotToFile()
 
         unlink($screenshotPath);
     }
+
+    /**
+     * @covers ::takeScreenshot
+     */
+    public function testShouldSaveScreenshotToFileInSubDirectory()
```
Comment (OndraM) [inline tests/functional/RemoteWebDriverTest.php]:
```suggestion
    public function testShouldSaveScreenshotToFileAndCreateDirectory()
```

**Comment by OndraM** [comment]
Also, if you rebase the tests against community branch, the Travis builds should now be fixed.

**Comment by SjorsO** [comment]
@OndraM I've implemented your feedback, let me know if there is anything else

**Comment by OndraM** [comment]
Merged manually, thanks!

**Pull request closed without merging**